addons/website/static/src/scss/website.scss

```diff
@@ -1049,26 +1050,27 @@ header {
                     display: none;
                 }
             }
-
-            .o_connected_user:not(.editor_has_snippets):not(.o_fullscreen) header & {
-                // TODO this has to be reviewed
-                $-header-template: o-website-value('header-template');
-                @if not index(('Contact', 'boxed', 'magazine'), $-header-template) {
-                    top: -$o-navbar-height;
-                    padding-top: $o-navbar-height;
-                } @else if $-header-template == 'magazine' {
-                    @include media-breakpoint-up(lg) {
-                        padding-top: $o-navbar-height;
-                    }
+            @if o-website-value('header-template') == 'vertical' {
+                .oe_structure_solo {
+                    // Hide the "oe_structure_solo" because it is outside the
+                    // menu when the off-canvas menu is open.
+                    display: none;
                 }
             }
-
-            .editor_has_snippets:not(.o_fullscreen) header & {
-                @if (o-website-value('header-template') == 'magazine') {
-                    @include media-breakpoint-up(lg) {
-                        right: $o-we-sidebar-width;
-                    }
-                }
+            .o_connected_user:not(.editor_has_snippets):not(.o_fullscreen) header & {
+                padding-top: $o-navbar-height;
+            }
+            // The "off-canvas" navbar has a "fixed" position (therefore
+            // relative to the viewport). But when the header is "affixed", the
+            // "fixed" position of the navbar is no longer relative to the
+            // viewport but it is relative to the header (this is due to
+            // "transform" CSS rule of the "affixed" header.
+            .o_connected_user:not(.editor_has_snippets):not(.o_fullscreen) header.o_header_affixed & {
+                top: -$o-navbar-height;
+            }
+            .editor_has_snippets:not(.o_fullscreen) header:not(.o_header_affixed) &.collapsing,
+            .editor_has_snippets:not(.o_fullscreen) header:not(.o_header_affixed) &.show {
+                right: $o-we-sidebar-width;
```

Comment:
I retested everything to figure out how to resolve the conflict. It should be good. :pray::pray::pray: